store/buildSkyblockCalendar.js

```diff
@@ -0,0 +1,309 @@
+}
+
+function getJacobEventTimes() {
+  const times = [];
+
+  for (let month = 0; month < 12; month++) {
+    let day = 2;
+    while (day <= 31) {
+      times.push({
+        start: getOffset(months[month], day),
+        end: getOffset(months[month], day),
+      });
+
+      if (day === 30) break;
+      day += 3;
+      if (day > 31) {
+        day %= 31;
+        month++;
+      }
+    }
+  }
+  return times;
+}
+
+function getUniqueListBy(array, key) {
+  return [...new Map(array.map((item) => [item[key], item])).values()];
+}
+
+const eventTimes = {
+  BANK_INTEREST: {
+    name: 'Bank Interest',
+    times: [
+      {
+        start: getOffset('Early Spring', 1),
+        end: getOffset('Early Spring', 1),
+      },
+      {
+        start: getOffset('Early Summer', 1),
+        end: getOffset('Early Summer', 1),
+      },
+      {
+        start: getOffset('Early Autumn', 1),
+        end: getOffset('Early Autumn', 1),
+      },
+      {
+        start: getOffset('Early Winter', 1),
+        end: getOffset('Early Winter', 1),
+      },
+    ],
+  },
+  ELECTION_OVER: {
+    name: 'Election Over',
+    times: [
+      {
+        start: getOffset('Late Spring', 27),
+        end: getOffset('Late Spring', 27),
+      },
+    ],
+  },
+  TRAVELING_ZOO: {
+    name: 'Traveling Zoo',
+    times: [
+      {
+        start: getOffset('Early Summer', 1),
+        end: getOffset('Early Summer', 3),
+      },
+      {
+        start: getOffset('Early Winter', 1),
+        end: getOffset('Early Winter', 3),
+      },
+    ],
+  },
+  ELECTION_BOOTH_OPENS: {
+    name: 'Election Booth Opens',
+    times: [
+      {
+        start: getOffset('Late Summer', 27),
+        end: getOffset('Late Summer', 27),
+      },
+    ],
+  },
+  FEAR_MONGERER: {
+    name: 'Fear Mongerer',
+    times: [
+      {
+        start: getOffset('Autumn', 26),
+        end: getOffset('Late Autumn', 3),
+      },
+    ],
+  },
+  SPOOKY_FESTIVAL: {
+    name: 'Spooky Festival',
+    times: [
+      {
+        start: getOffset('Autumn', 29),
+        end: getOffset('Autumn', 31),
+      },
+    ],
+  },
+  JERRYS_WORKSHOP: {
+    name: 'Jerry\'s Workshop',
+    times: [
+      {
+        start: getOffset('Late Winter', 1),
+        end: getOffset('Late Winter', 31),
+      },
+    ],
+  },
+  SEASON_OF_JERRY: {
+    name: 'Season of Jerry',
+    times: [
+      {
+        start: getOffset('Late Winter', 24),
+        end: getOffset('Late Winter', 26),
+      },
+    ],
+  },
+  NEW_YEAR_CELEBRATION: {
+    name: 'New Year Celebration',
+    times: [
+      {
+        start: getOffset('Late Winter', 29),
+        end: getOffset('Late Winter', 31),
+      },
+    ],
+  },
+  JACOBS_CONTEST: {
+    name: 'Jacob\'s Farming Contest',
+    times: getJacobEventTimes(),
+  },
+};
+
+function buildSkyblockCalendar(events = '', from = Date.now(), to, years, stopAtYearEnd = false) {
+  if (Number.isNaN(Number(to)) && Number.isNaN(Number(years))) {
```

Comment:
The `from` and `to` parameter should use the custom format (i.e. call ``parseTimestamp()) described here https://github.com/slothpixel/core/wiki/Using-custom-date-parameters

Also querying only the `to` parameter causes an error

Reply:
For some reason, the `to` parameter currently works for me but I'll change it to match it as you said.

Reply:
for the `from` parameter, do you want it to error out if its below or just set it to `yearZero` if its below?

Reply:
I think just setting to `yearZero` is fine